pkgs/development/mobile/androidenv/examples/shell.nix

```diff
@@ -138,11 +138,157 @@ pkgs.mkShell rec {
 
     # Write out local.properties for Android Studio.
     cat <<EOF > local.properties
-# This file was automatically generated by nix-shell.
-sdk.dir=$ANDROID_SDK_ROOT
-ndk.dir=$ANDROID_NDK_ROOT
-cmake.dir=$cmake_root
-EOF
+    # This file was automatically generated by nix-shell.
+    sdk.dir=$ANDROID_SDK_ROOT
+    ndk.dir=$ANDROID_NDK_ROOT
+    cmake.dir=$cmake_root
+    EOF
   '';
+
+  passthru.tests = {
+    sdkmanager-licenses-test = pkgs.runCommand "sdkmanager-licenses-test" {
+      buildInputs = [ androidSdk jdk ];
+    } ''
+      if [[ ! $(sdkmanager --licenses) =~ "All SDK package licenses not accepted." ]]; then
+        echo "At least one of SDK package licenses are not accepted."
+        exit 1
+      fi
+      touch $out
+    '';
+
+    sdkmanager-packages-test = pkgs.runCommand "sdkmanager-packages-test" {
+      buildInputs = [ androidSdk jdk ];
+    } ''
+      output=$(sdkmanager --list)
+      installed_packages_section=$(echo "''${output%%Available Packages*}" | awk 'NR>4 {print $1}')
+
+      if [[ ! $installed_packages_section =~ "build-tools;30.0.3" ]]; then
```

Comment:
Can you do something like:

```
for section in 'ndk-bundle' 'platform-tools' 'platforms;android-23' ...; do
  if [[ ! "$installed_packages_section =~ "$section" ]; then
    echo "$section was not installed"
  fi
done
```

Reply:
fixed on [a39b895](https://github.com/NixOS/nixpkgs/pull/208793/commits/a39b895f77668a0b58b53702729b9d5d7101963d)